packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/summary.js

```diff
@@ -17,72 +18,104 @@ const Summary = props => {
-                    />
-                    <span className={classes.edit_text}>{'Edit'}</span>
-                </button>
-            </div>
-            <div className={classes.card_details_container}>
-                <span className={classes.payment_type}>Credit Card</span>
-                <span className={classes.payment_details}>
-                    {paymentSummary}
-                </span>
+        const billingAddressSummary =
+            !isBillingAddressSame && billingAddress ? (
+                <div className={classes.address_summary_container}>
+                    <div>
+                        <span className={classes.first_name}>
+                            {billingAddress.firstName}
+                        </span>
+                        <span className={classes.last_name}>
+                            {billingAddress.lastName}
+                        </span>
+                    </div>
+                    <div>
+                        <span className={classes.street1}>
+                            {billingAddress.street1}
+                        </span>
+                        <span className={classes.street2}>
+                            {billingAddress.street2}
+                        </span>
+                        <span className={classes.city}>
+                            {billingAddress.city}
+                        </span>
+                        <span className={classes.state}>
+                            {billingAddress.state}
+                        </span>
+                    </div>
+                    <div>
+                        <span className={classes.postalCode}>
+                            {billingAddress.postalCode}
+                        </span>
+                        <span className={classes.country}>
+                            {billingAddress.country}
+                        </span>
+                    </div>
+                </div>
+            ) : null;
+
+        return (
+            <div className={classes.root}>
+                <div className={classes.heading_container}>
+                    <h5 className={classes.heading}>Payment Information</h5>
+                    <button
+                        className={classes.edit_button}
+                        onClick={onEdit}
+                        type="button"
+                    >
+                        <Icon
+                            size={16}
+                            src={EditIcon}
+                            classes={{ icon: classes.edit_icon }}
+                        />
+                        <span className={classes.edit_text}>{'Edit'}</span>
+                    </button>
+                </div>
+                <div className={classes.card_details_container}>
+                    <span className={classes.payment_type}>Credit Card</span>
+                    <span className={classes.payment_details}>
+                        {paymentSummary}
```

Comment:
Probably not related here/out of scope for this PR, but it would be great if we can separate out summary just like how you have done with payment methods to make it easy for future developers to add new summary components. For instance: https://github.com/magento/pwa-studio/pull/2409/files#diff-25736d2fa0849cf5b71064fe3e09a608

Reply:
When we implement additional methods this will become a necessity but definitely out of scope for now I think.